src/convert/impls.rs

```diff
@@ -624,3 +624,163 @@ where
     }
     result.into_boxed_slice()
 }
+
+// WebAssembly ToWebAssemblyValue implementations for primitive types
+// These implement the exact WebAssembly specification algorithms
+
+impl TryFromJsValue for i32 {
+    type Error = JsError;
+
+    fn try_from_js_value(js_value: JsValue) -> Result<Self, Self::Error> {
+        // WebAssembly ToInt32(v) algorithm
+        let number = unsafe { crate::__wbindgen_as_number(js_value.idx) };
+        let result = if !number.is_finite() || number == 0.0 || number == -0.0 {
+            0i32
+        } else {
+            let int = number as i64;
+            let int32bit = (int as u32) as i64;
+            if int32bit >= 1_i64 << 31 {
+                (int32bit - (1_i64 << 32)) as i32
+            } else {
+                int32bit as i32
+            }
+        };
+        Ok(result)
+    }
+}
+
+impl TryFromJsValue for u32 {
+    type Error = JsError;
+
+    fn try_from_js_value(js_value: JsValue) -> Result<Self, Self::Error> {
+        // WebAssembly ToUint32(v) algorithm
+        let number = unsafe { crate::__wbindgen_as_number(js_value.idx) };
+        let result = if !number.is_finite() || number == 0.0 || number == -0.0 {
+            0u32
+        } else {
+            let int = number as i64;
+            int as u32
+        };
+        Ok(result)
+    }
+}
+
+impl TryFromJsValue for i64 {
+    type Error = JsError;
+
+    fn try_from_js_value(js_value: JsValue) -> Result<Self, Self::Error> {
+        // WebAssembly ToBigInt64(v) algorithm
+        crate::bigint_get_as_i64(&js_value)
+            .ok_or_else(|| JsError::new("Failed to convert to BigInt64"))
+    }
+}
+
+impl TryFromJsValue for u64 {
+    type Error = JsError;
+
+    fn try_from_js_value(js_value: JsValue) -> Result<Self, Self::Error> {
+        // WebAssembly ToBigUint64(v) algorithm
+        crate::bigint_get_as_i64(&js_value)
+            .map(|i64_val| i64_val as u64)
```

Comment:
Should probably also use `try_into()` if it's meant to be fallible.

Reply:
I'm refactoring these to rather be a new infallible trait implementation under the assumption that type system violations are panics by definition.